build.sbt

```diff
@@ -161,51 +165,56 @@ val scalacJvmOptions = Def.setting {
-  nativeImageVersion := "22.3.0",
-  nativeImageInstalled := isCI,
-  nativeImageOptions ++= {
-    val isMusl = sys.env.get("NATIVE_IMAGE_MUSL").exists(_.toBoolean)
-    if (isMusl) Seq("--libc=musl") else Nil
-  },
-  nativeImageOptions ++= {
-    val isStatic = sys.env.get("NATIVE_IMAGE_STATIC").exists(_.toBoolean)
-    if (isStatic) Seq("--static") else Nil
-  },
-).dependsOn(coreJVM, dynamic).enablePlugins(NativeImagePlugin)
+lazy val cli = crossProject(JVMPlatform).withoutSuffixFor(JVMPlatform)
+  .in(file("scalafmt-cli")).settings(
+    moduleName := "scalafmt-cli",
+    assembly / mainClass := Some("org.scalafmt.cli.Cli"),
+    assembly / assemblyOption := (assembly / assemblyOption).value
+      .withPrependShellScript(Some(defaultUniversalScript(shebang = false))),
+    assembly / assemblyJarName := "scalafmt.jar",
+    assembly / assemblyMergeStrategy := {
+      case "reflect.properties" => MergeStrategy.first
+      case x =>
+        val oldStrategy = (assembly / assemblyMergeStrategy).value
+        oldStrategy(x)
+    },
+    libraryDependencies ++= Seq(
+      "org.scalameta" %% "munit-diff" % "1.0.2",
+      "com.martiansoftware" % "nailgun-server" % "0.9.1",
+      "com.github.scopt" %% "scopt" % "4.1.0",
+    ),
+    scalacOptions ++= scalacJvmOptions.value,
+    Compile / mainClass := Some("org.scalafmt.cli.Cli"),
+    nativeImageVersion := "22.3.0",
+    nativeImageInstalled := isCI,
+    nativeImageOptions ++= {
+      val isMusl = sys.env.get("NATIVE_IMAGE_MUSL").exists(_.toBoolean)
+      if (isMusl) Seq("--libc=musl") else Nil
+    },
+    nativeImageOptions ++= {
+      val isStatic = sys.env.get("NATIVE_IMAGE_STATIC").exists(_.toBoolean)
+      if (isStatic) Seq("--static") else Nil
+    },
+  ).dependsOn(core, dynamic).enablePlugins(NativeImagePlugin)
 
-lazy val tests = project.in(file("scalafmt-tests")).settings(
-  publish / skip := true,
-  libraryDependencies ++= Seq(
-    // Test dependencies
-    "com.lihaoyi" %% "scalatags" % "0.13.1",
-    scalametaTestkit,
-    munit.value,
-  ),
-  scalacOptions ++= scalacJvmOptions.value,
-  javaOptions += "-Dfile.encoding=UTF8",
-  buildInfoPackage := "org.scalafmt.tests",
-  buildInfoKeys :=
-    Seq[BuildInfoKey]("resourceDirectory" -> (Test / resourceDirectory).value),
-).enablePlugins(BuildInfoPlugin).dependsOn(coreJVM, dynamic, cli)
+lazy val tests = crossProject(JVMPlatform).withoutSuffixFor(JVMPlatform)
+  .in(file("scalafmt-tests")).settings(
+    publish / skip := true,
+    libraryDependencies ++= Seq(
+      // Test dependencies
+      "com.lihaoyi" %% "scalatags" % "0.13.1",
+      scalametaTestkit,
+      munit.value,
+    ),
+    scalacOptions ++= scalacJvmOptions.value,
+    javaOptions += "-Dfile.encoding=UTF8",
+    buildInfoPackage := "org.scalafmt.tests",
+    buildInfoKeys := Seq[BuildInfoKey](
+      "resourceDirectory" -> (Test / resourceDirectories).value.find(
+        _.toPath.endsWith(Paths.get("shared") / "src" / "test" / "resources"),
```

Comment:
both approaches feel like a hack, especially this `src/test/resources` part. i think @tgodzik may have meant to look for the entry which *contains* `shared`, with the assumption that there's only one such entry so you don't need to specify what comes after `shared`.

Reply:
But then if someone clones scalafmt into some other `shared` directory, they are going to be very confused

Reply:
Think `/Users/shared/scalafmt/...`, etc.

Reply:
Also FormatTest throws an error if it cannot find any test files in the testDir, so it's not like we risk something being overlooked later on if we specify the test file location manually.

Reply:
I'm not an expert in sbt but gpt4 is, and it suggested using `(baseDirectory in ThisBuild).value` followed by `shared`, that should be unambiguous.

Reply:
also, does it make sense to add a new project specifically for shared?

Reply:
(baseDirectory in ThisBuild).value points to the top level `scalafmt` directory, and I need it to point to`scalafmt/scalafmt-tests/shared/src/test/resources` (so I could write `(ThisBuild / baseDirectory).value / "scalafmt-tests" / "shared" / "src" / "test" / "resources"`, but that is much different from my original proposal

Reply:
ok, then, more specifically: would something like this work
```
"resourceDirectory" -> {
  val sharedTests = (tests / baseDirectory).value / "shared"
  (Test / resourceDirectories).value.find(_.toPath.startsWith(sharedTests))
}
```
here, we explicitly use sbt constructs for everything except the constant `shared` in one place.

Reply:
Ah, ok, sure